app/services/vendor_api/generate_test_applications_for_provider.rb

```diff
@@ -1,9 +1,9 @@
 module VendorAPI
   class GenerateTestApplicationsForProvider
-    def call(provider:, courses_per_application:, count:, for_ratified_courses: false)
+    def call(provider:, courses_per_application:, count:, for_ratified_courses: false, mixed_application: false)
       raise ParameterInvalid, 'Parameter is invalid (cannot be zero): courses_per_application' if courses_per_application.zero?
 
-      course_ids = course_list_for_provider(provider, for_ratified_courses).pluck(:id)
+      course_ids = course_list_for_provider(provider, for_ratified_courses, mixed_application).pluck(:id)
```

Comment:
Im unsure around allowing both those values to be set, shouldn't it be either or, as it becomes confusing what takes precedence 🤔

Reply:
As discussed, we will add parameters to additively specify which courses we should choose from:
- for_ratified_courses
- for_training_courses
- for_test_provider_courses

Open to suggestions for names of these